modules/imgproc/test/test_cvtyuv.cpp

```diff
@@ -724,4 +724,13 @@ INSTANTIATE_TEST_CASE_P(cvt422, Imgproc_ColorYUV,
                       (int)COLOR_YUV2RGBA_YUY2, (int)COLOR_YUV2BGRA_YUY2, (int)COLOR_YUV2RGBA_YVYU, (int)COLOR_YUV2BGRA_YVYU,
                       (int)COLOR_YUV2GRAY_UYVY, (int)COLOR_YUV2GRAY_YUY2));
 
-}} // namespace
+}
+
+TEST(cvtColorUYVY, size_issue_21035)
+{
+    Mat input = Mat::zeros(1, 1, CV_8UC1);
```

Comment:
Here you pass `CV_8UC1` Mat as input, but I it is rejected early, before width check, because only 2 channel arrays are permitted.

Does it make sense to check multiple `cn` sizes?

Reply:
Fixed, used 8uc2.